app/policies/sample_policy.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+# Policy samples authorization
+class SamplePolicy < ApplicationPolicy
+  scope_for :relation, :group_samples do |relation, options|
+    group = options[:group]
```

Comment:
**nitpick (blocking):** Can you add the following after this line, to only return `Samples` when the `User` has permission, just as another failsafe.

```ruby
next scope.none if !Member.can_view?(user, group)
```

Reply:
Good call. Updated in 2c93588ea584bafca3f264f876280279799314d1